SerialPort in Galileo Gen 2 running Yocto

I'm trying to set a Serial Communication in a Intel Galileo Gen 2 with Linux Yocto, but I only get the message "**Illegal Instruction**".

Issue opened before in the wrong place: https://github.com/rwaldron/johnny-five/issues/1086

The code:

```
var Serialport = require("serialport").SerialPort;
var sp = new Serialport("/dev/ttyS0", {
  baudRate: 57600
});

sp.on("open", function() {
  console.log("Port is open!");

  // Once the port is open, you may read or write to it.
  sp.on("data", function(data) {
    console.log("Received: ", data);
  });

  sp.write(new Buffer("0120000003"));
});
```

Already tried to use the last beta version of SerialPort.

Something's wrong on the Galileo 2. Here's the output of `npm install serialport-galileo-test`
```
root@galileo:~/serialport-galileo-test# npm install serialport-galileo-test
-
> serialport-galileo-test@3.1.1-beta1 install /home/root/serialport-galileo-test/node_modules/serialport-galileo-test
> node-pre-gyp install --fallback-to-build


module.js:340
    throw err;
          ^
Error: Cannot find module 'nopt'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/root/serialport-galileo-test/node_modules/serialport-galileo-test/node_modules/node-pre-gyp/lib/node-pre-gyp.js:14:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

npm ERR! serialport-galileo-test@3.1.1-beta1 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the serialport-galileo-test@3.1.1-beta1 install script.
npm ERR! This is most likely a problem with the serialport-galileo-test package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls serialport-galileo-test
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.8.7-yocto-standard
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install" "serialport-galileo-test"
npm ERR! cwd /home/root/serialport-galileo-test
npm ERR! node -v v0.10.35
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
root@galileo:~/serialport-galileo-test# 
```

See also: https://github.com/voodootikigod/node-serialport/issues/780#issuecomment-217019406

--- a/README.md
+++ b/README.md
@@ -36,6 +36,7 @@ For getting started with node-serialport, we recommend you begin with the follow
   * [Ubuntu Linux](#ubuntu-linux)
   * [Alpine Linux](#alpine-linux)
   * [Raspberry Pi Linux](#raspberry-pi-linux)
+  * [Illegal Instruction](#illegal-instruction)
 * [Usage](#usage)
   * [Opening a Port](#opening-a-port)
   * [Listing Ports](#listing-ports)
@@ -136,6 +137,21 @@ npm install serialport --build-from-source=serialport
 #### Raspberry Pi Linux
 
 Follow the instructions for [setting up a Raspberry pi for use with Johnny-Five and Raspi IO](https://github.com/nebrius/raspi-io/wiki/Getting-a-Raspberry-Pi-ready-for-NodeBots). These projects use Node Serialport under the hood.
+
+#### Illegal Instruction
+
+The pre-compiled binaries assume a fully capable chip. The Galileo 2 for example lacks a few instruction sets from the `ia32` architecture. A few other platforms have similar issues. So if you get `Illegal Instruction` when trying to run serialport you'll need to rebuild the serialport binary by asking npm to rebuild it.
+
+```bash
+# Will ask npm to build serialport during install time
+npm install serialport --build-from-source
+
+# If you have a package that depends on serialport you can ask npm to rebuild it specifically.
+npm rebuild serialport --build-from-source
+
+# Or leave out the package name to rebuild everything.
+npm rebuild --build-from-source
+```
 
 ## Usage
 

